substrate/frame/election-provider-multi-block/src/lib.rs

```diff
@@ -2132,6 +2149,77 @@ mod phase_rotation {
 				assert_eq!(MultiBlock::current_phase(), Phase::Signed(SignedPhase::get() - 3));
 			});
 	}
+
+	#[test]
+	fn export_phase_blocks_on_initialize_transitions() {
+		ExtBuilder::full()
+			.pages(3)
+			.election_start(13)
+			.fallback_mode(FallbackModes::Onchain)
+			.build_and_execute(|| {
+				roll_to_done();
+
+				assert_eq!(MultiBlock::current_phase(), Phase::Done);
+
+				// Start export by calling elect(max_page)
+				assert_ok!(MultiBlock::elect(2)); // max_page = 2 for 3 pages
+				assert_eq!(MultiBlock::current_phase(), Phase::Export(1));
+
+				let current_block = System::block_number();
+
+				// Test that on_initialize does NOT advance the phase when in Export
+				roll_next();
+				assert_eq!(
+					MultiBlock::current_phase(),
+					Phase::Export(1),
+					"Export phase should not auto-transition"
+				);
+
+				// Only elect() should advance the Export phase
+				assert_ok!(MultiBlock::elect(1));
+				assert_eq!(MultiBlock::current_phase(), Phase::Export(0));
+
+				// Test Export(0) also blocks on_initialize transitions
+				MultiBlock::on_initialize(current_block + 4);
```

Comment:
```suggestion
				MultiBlock::on_initialize(current_block + 4);
```

This doesn't go forward 4 blocks FWIW, you want to do `roll_many(4)`.